core/src/token_info/cached.rs

```diff
@@ -151,4 +173,38 @@ mod tests {
             .expect("First fetch failed");
         assert_eq!(info, hardcoded);
     }
+
+    #[test]
+    fn all_ids_works() {
+        fn token_ids() -> Vec<TokenId> {
+            [0, 1, 2].iter().cloned().map(TokenId).collect()
+        }
+
+        let mut inner = MockTokenInfoFetching::new();
+        inner
+            .expect_all_ids()
+            .times(1)
+            .returning(|| immediate!(Ok(token_ids())));
+        inner
```

Comment:
Would be nice to test the documented behavior that it doesn't of one of the subqueries fails.